### ISSUE TITLE
Crashing on requested analysis for uid-column

### ISSUE THREAD

**Issue by sebastian** [diffix/explorer#186]
After the API change `air` sends over all the columns of a table.
This also includes the `uid` column. 
When querying the `uid` column `air` produces an error informing the user that selecting the `uid` column in the top-most query will not yield meaningful results. This seems to crash the explorer analysis.

```
fail: Microsoft.AspNetCore.Server.Kestrel[13]
      Connection id "0HM13Q8FEA7TO", Request id "0HM13Q8FEA7TO:00000001": An unhandled exception was thrown by the application.
System.Exception: Aircloak API query error.
DataSource: GamesAndPlayers.
Error: Directly selecting or grouping on the user id column in an anonymizing query is not allowed, as it would lead to a fully censored result.
Anonymizing queries can reference user id columns either by filtering on them, aggregating them, or processing them so that they have a chance to create anonymized buckets.

The error was detected at line 3, column 21:

	2:                    select
	3:                        player,
	                          ^
	4:                        count(*),
Query Id: f30d9e49-27ac-4da1-a72a-cdaf49b450ff
Query Statement:
                select
                    player,
                    count(*),
                    count_noise(*)
                from games
                group by player
   at Aircloak.JsonApi.JsonApiClient.PollQueryUntilComplete[TRow](Uri apiUrl, String queryId, DQuery`1 query, TimeSpan pollFrequency, CancellationToken cancellationToken) in /src/aircloak/JsonApi/JsonApiClient.cs:line 184
   at Aircloak.JsonApi.JsonApiClient.PollQueryUntilComplete[TRow](Uri apiUrl, String queryId, DQuery`1 query, TimeSpan pollFrequency, CancellationToken cancellationToken) in /src/aircloak/JsonApi/JsonApiClient.cs:line 202
   at Aircloak.JsonApi.JsonApiClient.Query[TRow](Uri apiUrl, String dataSource, DQuery`1 query, TimeSpan pollFrequency, CancellationToken cancellationToken) in /src/aircloak/JsonApi/JsonApiClient.cs:line 92
   at Aircloak.JsonApi.AircloakConnection.Exec[TRow](DQuery`1 query) in /src/aircloak/JsonApi/AircloakConnection.cs:line 51
   at Explorer.Components.DistinctValuesComponent.Explore() in /src/explorer/Components/DistinctValuesComponent.cs:line 85
   at Explorer.Components.ExplorerComponent`1.<get_ResultAsync>b__2_0() in /src/explorer/Components/ExplorerComponent.cs:line 11
   at Explorer.Components.DistinctValuesComponent.YieldMetrics()+MoveNext() in /src/explorer/Components/DistinctValuesComponent.cs:line 73
   at Explorer.Components.PublisherComponent.PublishMetrics(MetricsPublisher publisher) in /src/explorer/Components/PublisherComponent.cs:line 14
   at Explorer.Components.PublisherComponent.PublishMetrics(MetricsPublisher publisher) in /src/explorer/Components/PublisherComponent.cs:line 14
   at Explorer.Api.ExplorationRegistry.Remove(Guid id) in /src/explorer.api/Exploration/ExplorationRegistry.cs:line 37
   at Explorer.Api.Controllers.ExploreController.Result(Guid explorationId) in /src/explorer.api/Controllers/ExploreController.cs:line 100
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
```

**Comment by AndreiBozantan** [comment]
Beside the changes made by @dandanlen in #190 (improved exception handling) I plan to also add a check in order to completely avoid the queries that fail on `user_id` columns.

**Comment by AndreiBozantan** [comment]
Closed with #212.
